pages/pipelines/defining_steps.md.erb

```diff
@@ -270,6 +270,24 @@ steps:
 ```
 {: codeblock-file="pipeline.yml"}
 
+## Dynamic pipeline templates
+
+If you need the ability to use pipelines from a central catalog, or enforce certain configuration rules, you can either use dynamic pipelines and the [`pipeline upload`](/docs/agent/v3/cli-pipeline#uploading-pipelines) command to make this happen, or [write custom plugins](/docs/plugins/writing) and share them across your organization.
+
+To use dynamic pipelines and the pipeline upload command, you'd make a pipeline that looks something like this:
+
+```yml
+steps:
+  - command: enforce-rules.sh | buildkite-agent pipeline upload
+    label: "\:pipeline\: Upload"
+```
+
+Where each team defines their steps in `team-steps.yml`, and your templating logic is in `enforce-rules.sh` (which can be written in any language that can pass YAML to the pipeline upload).
```

Comment:
```suggestion
Each team defines their steps in `team-steps.yml`. Your templating logic is in `enforce-rules.sh`, which can be written in any language that can pass YAML to the pipeline upload.
```

Reply:
I'm not strongly attached to this variant (save for changing "where" to "here" or omitting it altogether), but it felt right to take the info from the round brackets and put it prominently back into the main sentence.